src/hooks/Orderbook/useDrawOrderbook.ts

```diff
@@ -301,41 +298,31 @@ export const useDrawOrderbook = ({
     const empty: number[] = [];
     const removed: number[] = [];
     const updated: number[] = [];
+    const unchanged: number[] = [];
 
     prevData.current.forEach((row, idx) => {
       if (!row) {
         empty.push(idx);
         return;
       }
 
+      let animationType = OrderbookRowAnimationType.NEW;
+
       if (mapOfOrderbookPriceLevels?.[row.price] === 0) {
         removed.push(idx);
-        drawOrderbookRow({
-          ctx,
-          idx,
-          rowToRender: row,
-          animationType: OrderbookRowAnimationType.REMOVE,
-        });
-      } else if (mapOfOrderbookPriceLevels?.[row.price] === row?.size) {
-        drawOrderbookRow({
-          ctx,
-          idx,
-          rowToRender: data[idx],
-          animationType: OrderbookRowAnimationType.NONE,
-        });
+        animationType = OrderbookRowAnimationType.REMOVE;
+      } else if (mapOfOrderbookPriceLevels?.[row.price] === row.size) {
+        unchanged.push(idx);
+        animationType = OrderbookRowAnimationType.NONE;
       } else {
         updated.push(idx);
-        drawOrderbookRow({
-          ctx,
-          idx,
-          rowToRender: row,
-          animationType: OrderbookRowAnimationType.NEW,
-        });
       }
+
+      drawOrderbookRow({ ctx, idx, rowToRender: row, animationType });
     });
 
     setTimeout(() => {
-      [...empty, ...removed, ...updated].forEach((idx) => {
+      [...empty, ...removed, ...updated, ...unchanged].forEach((idx) => {
```

Comment:
Since we are included `unchanged` to this list, I don't think we need to use a loop with `clearRect`. I think we can just clear the entirety of the canvas since we will be redrawing every single row now.